go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -147,22 +146,7 @@ func (rs *rowStreamer) buildPlan() error {
 		return err
 	}
 
-	st, err := rs.se.GetTableForPos(fromTable, "")
-	if err != nil {
-		// There is a scenario where vstreamer's table state can be out-of-date, and this happens
-		// with vitess migrations, based on vreplication.
-		// Vitess migrations use an elaborate cut-over flow where tables are swapped away while traffic is
-		// being blocked. The RENAME flow is such that at some point the table is renamed away, leaving a
-		// "puncture"; this is an event that is captured by vstreamer. The completion of the flow fixes the
-		// puncture, and places a new table under the original table's name, but the way it is done does not
-		// cause vstreamer to refresh schema state.
-		// There is therefore a reproducible valid sequence of events where vstreamer thinks a table does not
-		// exist, where it in fact does exist.
-		// For this reason we give vstreamer a "second chance" to review the up-to-date state of the schema.
-		// In the future, we will reduce this operation to reading a single table rather than the entire schema.
-		rs.se.ReloadAt(context.Background(), replication.Position{})
-		st, err = rs.se.GetTableForPos(fromTable, "")
-	}
```

Comment:
I remember adding this after an exhausting debugging session. Very glad to see this removed!